### PR TITLE
Add brokerURL and better handle TimeoutException in QueryRunner

### PR DESCRIPTION
**Pull request by npawar**
Add support for QueryRunner with brokerURL 
Don't throw exception if reached configured timeout, instead proceed as normal to print final statictics

### CODE REVIEW

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java [hunk 1]
```diff
@@ -299,50 +302,53 @@ public static QuerySummary singleThreadedQueryRunner(PerfBenchmarkDriverConf con
     long reportStartTime = startTime;
     int numReportIntervals = 0;
     int numTimesExecuted = 0;
-    while (numTimesToRunQueries == 0 || numTimesExecuted < numTimesToRunQueries) {
-      for (String query : queries) {
-        if (timeout > 0 && System.currentTimeMillis() - startTimeAbsolute > timeout) {
-          LOGGER.warn("Timeout of {} sec reached. Aborting", timeout);
-          throw new TimeoutException("Timeout of " + timeout + " sec reached. Aborting");
-        }
+    try {
```
Comment (npawar) [inline pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:305]: there's no change here other than adding TimeoutException and catching it

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java [hunk 2]
```diff
@@ -416,42 +422,49 @@ public static QuerySummary multiThreadedQueryRunner(PerfBenchmarkDriverConf conf
...
+              reportAndClearStatistics(numQueriesExecuted, numExceptions, totalBrokerTime, totalClientTime,
+                  statisticsList);
+            }
           }
         }
+        numTimesExecuted++;
       }
-      numTimesExecuted++;
+    } catch (TimeoutException e) {
+      LOGGER.info("Time out reached. {}", e.getMessage());
```
Comment (richardstartin) [inline pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:467]: timeouts should not be omitted from the reported statistics

Comment (npawar) [reply pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:467]: It's not. Event after this change, you'll still get the message `Time out reached. Timeout of 300000 sec reached. Aborting`
Prior to this change, after this an exception was thrown and no final statistics were printed.
After this change, we stop gracefully ad print the final stats.

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java [hunk 3]
```diff
@@ -53,56 +52,58 @@ public class QueryRunner extends AbstractBaseCommand implements Command {
   private static final long NANO_DELTA = (long) 5E5;
   private static final String CLIENT_TIME_STATISTICS = "CLIENT TIME STATISTICS";
 
-  @CommandLine.Option(names = {"-mode"}, required = true,
-      description = "Mode of query runner (singleThread|multiThreads|targetQPS|increasingQPS).")
+  @CommandLine.Option(names = {"-mode"}, required = true, description = "Mode of query runner "
+      + "(singleThread|multiThreads|targetQPS|increasingQPS).")
```
Comment (Jackie-Jiang) [inline pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:56]: These changes do not seem to improve readability. IIRC we have disabled these kind of auto-reformat in the `Pinot Style`, can you try re-importing the style sheet?

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java [hunk 4]
```diff
@@ -295,15 +297,18 @@ public static QuerySummary singleThreadedQueryRunner(PerfBenchmarkDriverConf con
...
     int numReportIntervals = 0;
     int numTimesExecuted = 0;
-    while (numTimesToRunQueries == 0 || numTimesExecuted < numTimesToRunQueries) {
+
+    while (!timeoutReached && (numTimesToRunQueries == 0 || numTimesExecuted < numTimesToRunQueries)) {
       for (String query : queries) {
         if (timeout > 0 && System.currentTimeMillis() - startTimeAbsolute > timeout) {
-          LOGGER.warn("Timeout of {} sec reached. Aborting", timeout);
-          throw new TimeoutException("Timeout of " + timeout + " sec reached. Aborting");
+          LOGGER.info("Timeout of {} sec reached. Aborting", timeout);
```
Comment (Jackie-Jiang) [inline pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:309]: Suggest keeping it as warning. Same for other places

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java (hunk 3 above)
Comment (npawar) [reply pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:56]: just did, no changes

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java (hunk 4 above)
Comment (npawar) [reply pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:309]: imo this should be info. It is doing exactly what the config is asking it to do, and it ending here isn't an erroneous condition.

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java (hunk 3 above)
Comment (Jackie-Jiang) [reply pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:56]: The new one will honor the new lines and not try to merge separate lines, so if you revert and reformat, it won't apply the same change here. Not important though :-P

Comment (npawar) [reply pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:56]: ah i see! will keep in mind for next time

**Pull request merged**